### PR TITLE
Expand CUnit coverage for queue, extract and callbacklist subsystems

### PR DESCRIPTION
**Pull request by McWuragu**
### Motivation
- Increase unit-test coverage for existing subsystems to catch regressions and validate edge cases. 
- Add lightweight tests that require minimal or no mocking and exercise iterator/edge behaviors and parsing edge cases.

### Description
- Added new CUnit tests to `testsuit/queue_test.c` covering iterator reset behavior and deleting an empty queue, and updated `testsuit/include/queue_test.h` to register them. 
- Added new CUnit tests to `testsuit/extract_test.c` for missing-channel handling, `NULL` input handling, and direct channel extraction from payloads, and updated `testsuit/include/extract_test.h` to register them. 
- Added new CUnit tests to `testsuit/callbacklist_test.c` for case-insensitive tail search and head-removal link consistency, and updated `testsuit/include/callbacklist_test.h` to register them. 
- Only test code and test headers were changed; no production/source logic was modified and test suite counts were adjusted accordingly.

### Testing
- Attempted full autotools test pipeline with `autoreconf -fi && ./configure && make -j4 check`, which failed because `autopoint` is not available in the environment. 
- Attempted direct compilation with `gcc -Itestsuit/include -Isrc/include ... -lcunit -lpthread -o /tmp/testsuit_bin`, which failed because CUnit headers are not installed in the environment. 
- All changes were committed locally (`Expand CUnit coverage for queue, extract and callback subsystems`) and diffs show only test additions and header updates; no automated tests could be run successfully due to missing tooling/headers.

------
[Codex Task](https://chatgpt.com/codex/tasks/task_e_69a03bfaaa50832e8a7393b3d116a8fe)

### CODE REVIEW

File: testsuit/queue_test.c [hunk 1]
```diff
@@ -295,3 +295,23 @@ void test_getNextitrQueue_null_input(void) {
...
+
+void test_getNextitrQueue_resets_after_end(void) {
+    PQueue q = initQueue();
+    const char A[] = "A";
+
+    CU_ASSERT_EQUAL(pushQueue(q, make_qd(A, sizeof A)), QUEUE_SUCCESS);
+
+    CU_ASSERT_PTR_NOT_NULL(getNextitrQueue(q));
+    CU_ASSERT_PTR_NULL(getNextitrQueue(q));
+    CU_ASSERT_PTR_NOT_NULL(getNextitrQueue(q));
```
[the remark this answers is not in the document]
Comment (McWuragu) [reply testsuit/queue_test.c:327]: adjusted the test to the current api implementation
